src/spn/structure/leaves/histogram/Text.py

```diff
@@ -41,12 +41,32 @@ def histogram_to_str(node, feature_names=None, node_to_str=None):
 
 
 def histogram_tree_to_spn(tree, features, obj_type, tree_to_spn):
-    breaks = list(map(ast.literal_eval, tree.children[1].children))
-    densities = list(map(ast.literal_eval, tree.children[2].children))
-    bin_repr_points = list(map(ast.literal_eval, tree.children[3].children))
-    node = Histogram(breaks, densities, bin_repr_points)
+    # ToDo: Revert ASAP -- Parse-Tree structure has been changed for now; forced alternate processing. (2020-JAN-13)
+    # Revert by removing True-branch, but note that it might be necessary to account for other grammar changes.
+    if True:
```

Comment:
If I understand that correctly, the `True`-branch is only needed for the legacy benchmark files in text-format? 
In this case, I would prefer to have a additional argument to the function `histogram_tree_to_spn` which defaults to `False` and is used in the `if`-condition.

Reply:
> If I understand that correctly, the True-branch is only needed for the legacy benchmark files in text-format?

Yes.

> In this case, I would prefer to have a additional argument to the function histogram_tree_to_spn which defaults to False and is used in the if-condition.

Since *_tree_to_spn functions are called via dict lookups, one would have to check if the current nodetype (tnode) is "histogram" (See: [here](https://github.com/sommerlukas/SPFlow/pull/3/commits/3419bfee4a4da9c7555041bb9a14e2973a1e7c4c#diff-0466e83a2843032f017af0fc87092012R156)).
But, I found another (IMO better) solution.
Check "use-case", either string equation or ref-graph: [here](https://github.com/sommerlukas/SPFlow/pull/3/commits/44ca46ad2b934712d24859857c5265a53cfcfcef#diff-99f45c1faa106b884a3e0ac258735ba0R43-R48)
Check "if legacy" (tree_count would be 2): [here](https://github.com/sommerlukas/SPFlow/pull/3/commits/44ca46ad2b934712d24859857c5265a53cfcfcef#diff-99f45c1faa106b884a3e0ac258735ba0R55-R57)

This distinction also allowed to revert the output format ([here](https://github.com/sommerlukas/SPFlow/pull/3/commits/44ca46ad2b934712d24859857c5265a53cfcfcef#diff-99f45c1faa106b884a3e0ac258735ba0R39)).

So, now we're (partially -- i.e. as far as the xspn-benchmarks and some toy examples with histogram and categorical nodes are concerned) compatible to the current version of SPFlows to_str methods and will produce sound SPNs.

Reply:
Sounds good!